packages/plugin-http-client/src/rest/AbstractResource.ts

```diff
@@ -0,0 +1,170 @@
+import { GenericError, Dependencies } from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    REST_CLIENT_BASE_API_URL: string;
+  }
+}
+
+import {
+  HttpClient,
+  HttpClientRequestMethod,
+  HttpClientRequestOptions,
+} from '../HttpClient';
+
+export class AbstractResource {
+  #httpClient: HttpClient;
+  #baseApiUrl: string;
+
+  static get $dependencies(): Dependencies {
+    return [HttpClient, '?REST_CLIENT_BASE_API_URL'];
```

Comment:
volitelna dependency, ale v constructoru to bez ni hodi chybu?

Reply:
Slavek chtěl, aby to nespadlo "jen tak", ale vypsalo hezkou chybu - proto to tak bylo (volitelně a až custom error v constructoru). Nově jsem to hodil so $Settings.